Cineaste/Models/StoredMovie.swift

```diff
@@ -20,9 +32,129 @@ class StoredMovie: NSManagedObject {
+        case watched
+        case watchedDate
+        case id
+        case posterPath = "poster_path"
+        case title
+    }
+
+    required convenience init(from decoder: Decoder) throws {
+        guard let context = decoder.userInfo[.context] as? NSManagedObjectContext else {
+            throw StoredMovieDecodingError.findContext
+        }
+
+        let name = String(describing: type(of: self))
+        guard let entity = NSEntityDescription.entity(forEntityName: name, in: context) else {
+            throw StoredMovieDecodingError.entityCreation
+        }
+
+        self.init(entity: entity, insertInto: context)
+
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        id = try container.decode(Int64.self, forKey: .id)
+        title = try container.decode(String.self, forKey: .title)
+        overview = try container.decode(String.self, forKey: .overview)
+        posterPath = try container.decodeIfPresent(String.self, forKey: .posterPath)
+
+        voteAverage = try container.decode(Decimal.self, forKey: .voteAverage) as NSDecimalNumber
+        voteCount = try container.decode(Float.self, forKey: .voteCount)
+        runtime = try container.decode(Int16.self, forKey: .runtime)
+
+        let releaseDateString = try container.decode(String.self, forKey: .releaseDate)
+        guard let releaseDate = releaseDateString.dateFromImportedMoviesString else {
+            throw StoredMovieDecodingError.dateFromString
+        }
+        self.releaseDate = releaseDate
+
+        watched = try container.decode(Bool.self, forKey: .watched)
+
+        if let watchedDateString = try container.decodeIfPresent(String.self, forKey: .watchedDate) {
+            guard let watchedDate = watchedDateString.dateFromImportedMoviesString else {
+                throw StoredMovieDecodingError.dateFromString
+            }
+            self.watchedDate = watchedDate
+        }
+
+        listPosition = try container.decode(Int16.self, forKey: .listPosition)
+    }
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(overview, forKey: .overview)
+        try container.encode(listPosition, forKey: .listPosition)
+        try container.encode(releaseDate?.formattedForJson, forKey: .releaseDate)
+        try container.encode(runtime, forKey: .runtime)
+
+        if let voteAverage = voteAverage as Decimal? {
+            try container.encode(voteAverage, forKey: .voteAverage)
+        }
+
+        try container.encode(voteCount, forKey: .voteCount)
+        try container.encode(watched, forKey: .watched)
+        try container.encodeIfPresent(watchedDate?.formattedForJson, forKey: .watchedDate)
+        try container.encode(id, forKey: .id)
+        try container.encodeIfPresent(posterPath, forKey: .posterPath)
+        try container.encode(title, forKey: .title)
+    }
+
+}
+
+extension StoredMovie {
```

Comment:
This seems to break a bit with my idea to seperate all network related code from the persistence code. 
I know why it is comfortable to use this code but what do you think about making the load poster resource within the `Movie` entity static? This way we would reduce redundancy.

Reply:
Fixed that.